source/facter/3.0/release_notes.md

```diff
@@ -5,6 +5,81 @@ title: "Facter 3.0: Release Notes"
 
+Facter 3.0.2
+-----
+
+Released July 15, 2015.
+
+Shipped in puppet-agent version: 1.2.2.
+
+Facter 3.0.2 fixes several regressions in Facter 3.0.0 and implements new features to support legacy workflows.
+
+### Feature: `--show-legacy` Flag Outputs Unstructured Facts
+
+Facter 3 outputs new, structured versions of many facts, such as `os`, and hides the equivalent unstructured legacy facts. In Facter 3.0.2, you can force Facter's output to include legacy facts by running it with the `--show-legacy` flag.
+
+- [FACT-1075](https://tickets.puppetlabs.com/browse/FACT-1075)
+
+### Feature: Improved Memory Usage on Launch
+
+Previous versions of Facter could consume more memory than was necessary on a `puppetserver` launch. Facter 3.0.2 significantly improves memory usage on launch, making `puppetserver` easier to start on low-spec systems.
+
+- [FACT-1083](https://tickets.puppetlabs.com/browse/FACT-1083)
+
+### FIX: Improved Parsing of Kernel-derived Major and Minor Operating System Version Facts
+
+In Facter 3.0.2, the `os` fact reports major and minor kernel release versions on operating systems that do not formally report OS release versions, such as OS X and Arch Linux.
+
+- [FACT-1056: Operatingsystem major release not resolved in OSX](https://tickets.puppetlabs.com/browse/FACT-1056)
+- [FACT-1073: Operatingsystem release facts not properly resolved in Arch Linux](https://tickets.puppetlabs.com/browse/FACT-1073)
+
+### REGRESSION FIX: Restored `xendomains` Fact
+
+Facter 3.0.2 reports the `xendomains` fact, which was unintentionally omitted from Facter 3.0.0.
+
+- [FACT-867](https://tickets.puppetlabs.com/browse/FACT-867)
+
+### REGRESSION FIX: Restored Hostname to the `fqdn` Fact
+
+When reporting the `fqdn` fact, Facter 3.0.0 did not report the hostname. Facter 3.0.2 restores the hostname to the fact.
+
+- [FACT-1077](https://tickets.puppetlabs.com/browse/FACT-1077)
+- [FACT-1078](https://tickets.puppetlabs.com/browse/FACT-1078)
+
+### REGRESSION FIX: Fixed Crash on Windows Systems with Network Interfaces Lacking DHCP Assignments
+
+Facter 3.0.0 could crash with an "unhandled exception" error when reporting facts on a Windows system where DHCP-configured network interfaces weren't assigned IP addresses. Facter 3.0.2 resolves this issue.
+
+- [FACT-1084](https://tickets.puppetlabs.com/browse/FACT-1084)
+
+### REGRESSION FIX: Resolved an OLE COM Initialization Issue When Reporting Windows Facts 
+
+Facter 3.0.0 could fail when reporting Windows facts that require `win32ole`, resulting in a "fail: OLE initialize" error. Facter 3.0.2 resolves this issue.
+
+- [FACT-1082](https://tickets.puppetlabs.com/browse/FACT-1082)
+
+### REGRESSION FIX: Restored Reporting On Command Execution And Failures
+
+Facter 3.0.0 did not report output if a command it executed returned a non-zero exit code or couldn't be found. Facter 3.0.2 restores this functionality to Facter 2.x's behavior, producing error output to help diagnose fact-reporting failures.
```

Comment:
This was also used to produce facts that could only be found by causing a program to error, such as version info.